retriever.py

```diff
@@ -0,0 +1,272 @@
+               ["https://forecast.fz-juelich.de/campaigns2019",
+                "ecmwf.VS_ertel_potential_vorticity_ml", "ertel_potential_vorticity_bh"],
+               ["https://forecast.fz-juelich.de/campaigns2019",
+                "ecmwf.TroposphereInversionLayer", ""]
+       ]
+
+       will plot flight "ST25" with configured map section "01 SADPAP (stereo)" and
+       vertical range 500hPa to 50hPa from the given FTML file for init time
+       "2019-07-01T00:00:00Z" and valid time "2019-09-01T12:00:00Z". The plots
+       are defined in the hsecs (horizontal cross-sections) and vsecs (vertical
+       cross-sections) entries given each the URL of the server, the layer name, the style,
+       and, for hsec only, the elevation to plot (if necessary).
+    """)
+    parser.add_argument("-v", "--version", help="show version", action="store_true", default=False)
+    parser.add_argument("--debug", help="show debugging log messages on console", action="store_true", default=False)
+    parser.add_argument("--logfile", help="Specify logfile location. Set to empty string to disable.", action="store",
+                        default=os.path.join(mslib.msui.constants.MSUI_CONFIG_PATH, "mss_pyui.log"))
+    args = parser.parse_args()
+
+    if args.version:
+        print("***********************************************************************")
+        print("\n            Mission Support System (mss_retriever)\n")
+        print("***********************************************************************")
+        print("Documentation: http://mss.rtfd.io")
+        print("Version:", mslib.__version__)
+        sys.exit()
+
+    mslib.utils.setup_logging(args)
+
+    read_config_file()
+    config = config_loader()
+    print("o", config["automated_plotting_vsecs"])
+    print("OO")
+    print(config["predefined_map_sections"])
+    print(config["automated_plotting_flights"])
+
+    num_interpolation_points = config["num_interpolation_points"]
+    num_labels = config["num_labels"]
+    tick_index_step = num_interpolation_points // num_labels
+
+    fig = plt.figure()
+    for flight, section, vertical, filename, init_time, time in \
+            config["automated_plotting_flights"]:
+        params = mslib.utils.coordinate.get_projection_params(
+            config["predefined_map_sections"][section]["CRS"].lower())
+        params["basemap"].update(config["predefined_map_sections"][section]["map"])
+        wps = load_from_ftml(filename)
+        wp_lats, wp_lons, wp_locs = [[x[i] for x in wps] for i in [0, 1, 3]]
+        wp_presss = [mslib.utils.thermolib.flightlevel2pressure(wp[2] * units.hft).to("Pa").m for wp in wps]
+        for url, layer, style, elevation in config["automated_plotting_hsecs"]:
+            fig.clear()
+            ax = fig.add_subplot(111, zorder=99)
+            bm = mslib.msui.mpl_map.MapCanvas(ax=ax, **(params["basemap"]))
+
+            # plot path and labels
+            bm.plot(wp_lons, wp_lats,
```

Comment:
This is simplified, drastically, from 
mslib/msui/mpl_pathinteractor.py:1068ff

There is *probably* a crucial simplification here: The MSUI code plots the lines along great-circles (which is apparent for long sections on a cylindric projection). This here looks like "straight lines". If this is true, this needs to be fixed.